app/src/main/java/ch/epfl/culturequest/utils/ProfileUtils.java

```diff
@@ -35,34 +31,29 @@ public class ProfileUtils {
      * @param username the username to check
      * @return true if the username is valid, false otherwise
      */
-    public static boolean usernameIsValid(String username) {
+    private static USERNAME_PB usernameIsValid(String username) {
         int length = username.length();
-        return !username.isEmpty()
-                && length > 3
-                && length < 20
-                && username.matches(USERNAME_REGEX)
-                && !username.contains(" ");
+        USERNAME_PB result;
+        if (username.isEmpty())
+            result = USERNAME_PB.USERNAME_IS_EMPTY;
+        else if (length < 3)
+            result = USERNAME_PB.USERNAME_TOO_SHORT;
+        else if (length > 20)
```

Comment:
Maybe instead of only saying that it is too short or too long, it is better to say that it should be more than 3 character or less than 20, so that the user knows what too short or too long means